src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { px } from '../../../utils';
+import classNames from 'classnames';
+import { HeaderContext } from '../../Header/Header';
+
+export interface NavigationItemTriggerProps extends React.HTMLAttributes<HTMLElement> {
+  /**
+   * Label for the navigation item
+   */
+  label: string;
+}
+
+const NavigationItemTrigger = ({
+  label,
+  children,
+  className,
+  ...props
+}: React.PropsWithChildren<NavigationItemTriggerProps>) => {
+  const { expandedItem, handleSelection } = React.useContext(HeaderContext);
+  const handleOnClick = () => handleSelection(label);
+  return (
+    <li
+      className={classNames(className, `${px}-nav__item`, { [`${px}-nav__item--expanded`]: expandedItem === label })}
+      onClick={handleOnClick}
+    >
+      <button className={`${px}-nav__item-trigger`} type="button" aria-expanded={expandedItem === label} {...props}>
```

Comment:
I think we should always spread the `{...props}` onto the top-level HTML element